### PR TITLE
feat: seed the kernel random pool from the TPM

### PR DESCRIPTION
**Pull request by smira**
Use the TPM2 feature to provide high-quality random bytes.

### CODE REVIEW

File: internal/pkg/rng/tpm.go [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+		if len(resp.RandomBytes.Buffer) == 0 {
+			return fmt.Errorf("received zero random bytes from the TPM: %w", err)
+		}
+
+		if err = WriteEntropy(resp.RandomBytes.Buffer); err != nil {
+			return fmt.Errorf("error writing random pool entropy: %w", err)
+		}
+
+		remaining -= len(resp.RandomBytes.Buffer)
```
Comment (frezbo) [inline internal/pkg/rng/tpm.go:81]: so what's about crediting versus just seeding?

**Review by frezbo — APPROVED** [review]

File: internal/pkg/rng/tpm.go (hunk 1 above)
Comment (smira) [reply internal/pkg/rng/tpm.go:81]: systemd doesn't credit, so I thought we should do same

Comment (frezbo) [reply internal/pkg/rng/tpm.go:81]: yes, i didn't know the difference, finally read this: https://systemd.io/RANDOM_SEEDS/

**Pull request merged**